src/main/java/org/prebid/server/settings/HttpApplicationSettings.java

```diff
@@ -0,0 +1,210 @@
+package org.prebid.server.settings;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import io.vertx.core.Future;
+import io.vertx.core.http.HttpClient;
+import io.vertx.core.http.HttpClientResponse;
+import io.vertx.core.json.DecodeException;
+import io.vertx.core.json.Json;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+import org.apache.commons.collections4.CollectionUtils;
+import org.prebid.server.exception.PreBidException;
+import org.prebid.server.execution.Timeout;
+import org.prebid.server.settings.model.Account;
+import org.prebid.server.settings.model.HttpFetcherResponse;
+import org.prebid.server.settings.model.StoredRequestResult;
+import org.prebid.server.util.HttpUtil;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.concurrent.TimeoutException;
+import java.util.stream.Collectors;
+
+/**
+ * Implementation of {@link ApplicationSettings}.
+ * <p>
+ * Fetches an application settings from another service via HTTP protocol.
+ * <p>
+ * In order to enable caching and reduce latency for read operations {@link HttpApplicationSettings}
+ * can be decorated by {@link CachingApplicationSettings}.
+ * <p>
+ * Expected the endpoint to satisfy the following API:
+ * <p>
+ * GET {endpoint}?request-ids=req1,req2&imp-ids=imp1,imp2,imp3
+ * <p>
+ * This endpoint should return a payload like:
+ * <pre>
+ * {
+ *   "requests": {
+ *     "req1": { ... stored data for req1 ... },
+ *     "req2": { ... stored data for req2 ... },
+ *   },
+ *   "imps": {
+ *     "imp1": { ... stored data for imp1 ... },
+ *     "imp2": { ... stored data for imp2 ... },
+ *     "imp3": null // If imp3 is not found
+ *   }
+ * }
+ * </pre>
+ */
+public class HttpApplicationSettings implements ApplicationSettings {
+
+    private static final Logger logger = LoggerFactory.getLogger(HttpApplicationSettings.class);
+
+    private static final String REQUEST_IDS_PARAM = "request-ids";
+
+    // FIXME: uncomment when will be implemented obtaining by imp id
+    // private static final String IMP_IDS_PARAM = "imp-ids";
+
+    private HttpClient httpClient;
+    private String endpoint;
+    private String ampEndpoint;
+
+    public HttpApplicationSettings(HttpClient httpClient, String endpoint, String ampEndpoint) {
+        this.httpClient = Objects.requireNonNull(httpClient);
+        this.endpoint = HttpUtil.validateUrl(Objects.requireNonNull(endpoint));
+        this.ampEndpoint = HttpUtil.validateUrl(Objects.requireNonNull(ampEndpoint));
+    }
+
+    /**
+     * Not supported and returns failed result.
+     */
+    @Override
+    public Future<Account> getAccountById(String accountId, Timeout timeout) {
+        return Future.failedFuture(new PreBidException("Not supported"));
```

Comment:
Maybe worth mentioning in the documentation that HTTP settings backed can't be used if /auction will be utilized.